src/group_theory/submonoid.lean

```diff
@@ -65,66 +95,84 @@ lemma is_submonoid_Union_of_directed {ι : Type*} [hι : nonempty ι]
 λ ⟨n₁, h₁⟩ ⟨n₂, h₂⟩, ⟨n₁ + n₂, by simp only [pow_add, *]⟩
 
-lemma multiples.add_mem {x y z: β} : (y ∈ multiples x) → (z ∈ multiples x) → (y + z ∈ multiples x) :=
+/--The set of multiples of an element of an add_monoid is closed under addition. -/
+lemma multiples.add_mem {x y z : β} :
+  (y ∈ multiples x) → (z ∈ multiples x) → (y + z ∈ multiples x) :=
 @powers.mul_mem (multiplicative β) _ _ _ _
 attribute [to_additive] powers.mul_mem
 
-@[to_additive is_add_submonoid]
+/-- The set of natural powers of an element of a monoid α is a submonoid of α. -/
+@[to_additive is_add_submonoid "The set of natural multiples of an element of an add_monoid α is an add_submonoid of α."]
 instance powers.is_submonoid (x : α) : is_submonoid (powers x) :=
 { one_mem := powers.one_mem,
   mul_mem := λ y z, powers.mul_mem }
 
-@[to_additive is_add_submonoid]
+/-- A monoid is a submonoid of itself. -/
+@[to_additive is_add_submonoid "An add_monoid is an add_submonoid of itself."]
 instance univ.is_submonoid : is_submonoid (@set.univ α) := by split; simp
 
-@[to_additive is_add_submonoid]
+/-- The preimage of a submonoid under a monoid hom is a submonoid of the domain. -/
+@[to_additive is_add_submonoid "The preimage of an add_submonoid under an add_monoid hom is an add_submonoid of the domain."]
 instance preimage.is_submonoid {γ : Type*} [monoid γ] (f : α → γ) [is_monoid_hom f]
   (s : set γ) [is_submonoid s] : is_submonoid (f ⁻¹' s) :=
 { one_mem := show f 1 ∈ s, by rw is_monoid_hom.map_one f; exact is_submonoid.one_mem s,
   mul_mem := λ a b (ha : f a ∈ s) (hb : f b ∈ s),
     show f (a * b) ∈ s, by rw is_monoid_hom.map_mul f; exact is_submonoid.mul_mem ha hb }
 
-@[instance, to_additive is_add_submonoid]
+/-- The image of a submonoid under a monoid hom is a submonoid of the codomain. -/
+@[instance, to_additive is_add_submonoid "The image of an add_submonoid under an add_monoid hom is an add_submonoid of the codomain."]
 lemma image.is_submonoid {γ : Type*} [monoid γ] (f : α → γ) [is_monoid_hom f]
   (s : set α) [is_submonoid s] : is_submonoid (f '' s) :=
 { one_mem := ⟨1, is_submonoid.one_mem s, is_monoid_hom.map_one f⟩,
   mul_mem := λ a b ⟨x, hx⟩ ⟨y, hy⟩, ⟨x * y, is_submonoid.mul_mem hx.1 hy.1,
     by rw [is_monoid_hom.map_mul f, hx.2, hy.2]⟩ }
 
-@[to_additive is_add_submonoid]
+/-- The image of a monoid hom is a submonoid of the codomain. -/
+@[to_additive is_add_submonoid "The image of an add_monoid hom is an add_submonoid of the codomain."]
 instance range.is_submonoid {γ : Type*} [monoid γ] (f : α → γ) [is_monoid_hom f] :
   is_submonoid (set.range f) :=
 by rw ← set.image_univ; apply_instance
 
+/-- Submonoids are closed under natural powers. -/
 lemma is_submonoid.pow_mem {a : α} [is_submonoid s] (h : a ∈ s) : ∀ {n : ℕ}, a ^ n ∈ s
 | 0 := is_submonoid.one_mem s
 | (n + 1) := is_submonoid.mul_mem h is_submonoid.pow_mem
 
+/-- Add_submonoids are closed under multiplication by naturals. -/
```

Comment:
This is getting picky, so please feel free to ignore, but I wouldn't capitalise "code". Generally, I think it's best if identifiers which are actual Lean code get surrounded by backticks. Sometimes it's harder to write a sensible sentence, but given that the tooltips do actually typeset the backticks, it can be helpful. This could be:

"An `add_submonoid` is closed under multiplication by ...."